js/timers.ts

```diff
@@ -76,13 +83,20 @@ export function setInterval(
   // tslint:disable-next-line:no-any
   ...args: any[]
 ): number {
```

Comment:
I realize it's not from the Tokio stuff but was also the case before - but Is there any reason you prefer returning numbers from timers and not objects like Node does? Do you think the `.unref` stuff is not a concern/not interesting?

(Not related to the comments about maybe returning promises - which might not be 100% compatible but is certainly how it would have been done if they did these APIs today)

Reply:
No reason - actually I thought it was the spec.

Reply:
@ry well, I help maintain [lolex](https://github.com/sinonjs/lolex) so I dug into this quite a bit at times.

 - browsers "implement" the [Timers](https://www.w3.org/TR/2011/WD-html5-20110525/timers.html) spec - I say "implement" because browsers vary a lot between themselves in how they handle overflow and timer behaviour for edge cases.
 - Node doesn't implement that spec at all and returns objects from timers to expose optimizations.
 - Other platforms that run JavaScript don't do this or vary.

I think timer libs already expect either Node or browser timers mostly - we get some wiggle room - the important thing (other than timer ordering) is that `setTimeout` returns an object that `clearTimeout` clears.